src/EnvironmentChecker.php

```diff
@@ -120,6 +125,8 @@ public function init($permission = 'ADMIN')
                 "You must have the {$permission} permission to access this check"
             );
             throw new HTTPResponse_Exception($result);
+        } else {
+            $this->setIncludeDetails(true);
```

Comment:
This should be determined based on the route (`health/check` vs `dev/check` i.e. are we using the `DevHealthController` or the `DevCheckController`?) - not on the type and validity of authentication.

`setIncludeDetails()` is public, so I'd recommend calling it from `DevCheckController::index()`: https://github.com/silverstripe/silverstripe-environmentcheck/blob/d3d20e5539acc01687d16ed5b53ad408eedfbfbe/src/Controllers/DevCheckController.php#L46-L48